dask/dataframe/tests/test_groupby.py

```diff
@@ -296,6 +296,11 @@ def test_groupby_on_index(scheduler):
     pdf2 = pdf.set_index("a")
     assert_eq(ddf.groupby("a").b.mean(), ddf2.groupby(ddf2.index).b.mean())
 
+    # Check column projection for `groupby().agg`
+    agg = ddf2.groupby("a").agg({"b": "mean"})
+    assert_eq(ddf.groupby("a").b.mean(), agg.b)
+    assert hlg_layer(agg.dask, "getitem")
```

Comment:
Not meant as a blocking comment for this PR: I know we're using this check elsewhere to mean "we're doing column projection", but it feels somewhat indirect / brittle to me. For instance,`hlg_layer(agg.dask, "getitem")` means that there's _a_ `getitem` layer in the graph, but it could still pass even if we weren't getting column projection. Are there other things we can do that are a more direct check that we did not attempt to load in specific columns?

Reply:
Yeah, it is a shame how "indirect" these checks are.  I also hate that we are depending on the layers having an arbitrary name. For example, the layer we are checking for currently starts with "getitem", but "column-projection" or "series-projection/selection" could also make sense.

In the past, I rationalized these problems by recognizing that we **do** have more-thorough checks in `dataframe.io`  that confirm that `DataFrameIOLayer` -> `getitem` will result in column projections (and so all we really *need* here is to check that we have a `getitem` layer).  However, given the importance of column projection in `dask.dataframe`, it seems reasonable to add a new utility to explicitly return the set of columns that are being projected by the root [`DataFrameIOFunction`](https://github.com/dask/dask/blob/3f143db1a88af884d29a1b3678187c690ab641aa/dask/dataframe/io/utils.py#L202) of a DataFrame collection. Maybe something like:

```python
from dask.dataframe.utils import dataframe_io_columns

assert dataframe_io_columns(ddf) == {"name", "id"}
```

Reply:
Yeah, some utility that encapsulates a more direct column projection check would be nice 👍